ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -1,165 +1,408 @@
 #include "os.h"
 
 #include <cwchar>
+#include <irr/asset/filters/CSwizzleAndConvertImageFilter.h>
 
 #include "../../ext/MitsubaLoader/CMitsubaLoader.h"
 #include "../../ext/MitsubaLoader/ParserUtil.h"
 
-
 namespace irr
 {
+using namespace asset;
+
 namespace ext
 {
 namespace MitsubaLoader
 {
 
-static uint64_t rgb32f_to_rgb19e7(const uint32_t _rgb[3])
+_IRR_STATIC_INLINE_CONSTEXPR const char* DUMMY_VERTEX_SHADER =
+R"(#version 430 core
+
+layout (location = 0) in vec3 vPosition;
+layout (location = 2) in vec2 vUV;
+layout (location = 3) in vec3 vNormal;
+
+layout (location = 0) out vec3 WorldPos;
+layout (location = 1) flat out uint InstanceIndex;
+layout (location = 2) out vec3 Normal;
+layout (location = 3) out vec2 UV;
+
+#include <irr/builtin/glsl/utils/common.glsl>
+#include <irr/builtin/glsl/utils/transform.glsl>
+
+layout (push_constant) uniform Block {
+    uint instDataOffset;
+} PC;
+
+#ifndef _IRR_VERT_SET1_BINDINGS_DEFINED_
+#define _IRR_VERT_SET1_BINDINGS_DEFINED_
+layout (set = 1, binding = 0, row_major, std140) uniform UBO {
+    irr_glsl_SBasicViewParameters params;
+} CamData;
+#endif //_IRR_VERT_SET1_BINDINGS_DEFINED_
+
+struct InstanceData
 {
-	constexpr uint32_t mantissa_bitlen = 19u;
-	constexpr uint32_t exp_bitlen = 7u;
-	constexpr uint32_t exp_bias = 63u;
+	mat4x3 tform;
+	vec3 normalMatrixRow0;
+	uint bsdf_instrOffset;
+	vec3 normalMatrixRow1;
+	uint bsdf_instrCount;
+	vec3 normalMatrixRow2;
+	uint _padding;//not needed
+	uvec2 prefetch_instrStream;
+	uvec2 nprecomp_instrStream;
+	uvec2 genchoice_instrStream;
```

Comment:
I'm a bit confused about needing a `uvec2` for the rest of the streams

Reply:
same as https://github.com/buildaworldnet/IrrlichtBAW/pull/424#discussion_r509236731